src/Enzyme.jl

```diff
@@ -412,7 +414,9 @@ f(x) = x*x
     tt    = Tuple{map(T->eltype(Core.Typeof(T)), args)...}
     world = codegen_world_age(Core.Typeof(f.val), tt)
 
-    thunk = Enzyme.Compiler.thunk(Val(world), FA, RT, tt′, #=Mode=# Val(API.DEM_ForwardMode), Val(width),
+    opt_mi = RABI <: NonGenABI ? Compiler.fspec(eltype(FA), tt′) : Val(world)
```

Comment:
This observation of world is also an issue.

Reply:
But it shouldn't occur for the nongen case, no ?

Or do I need to literally separate into two functionsfor the different ABI's?

Reply:
I would write it as:

```
    opt_mi = if RABI <: NonGenABI
        Compiler.fspec(eltype(FA), tt′)
    else
        Val(codegen_world_age(Core.Typeof(f.val), tt))
    end
```

To ensure that world is not accidentally captured. And then look at all uses of `codegen_world_age` and do the same.